spectator/stateless_meters.h

```diff
@@ -137,12 +137,20 @@ class DistributionSummary : public StatelessMeter<Pub> {
 template <typename Pub>
 class Gauge : public StatelessMeter<Pub> {
  public:
-  Gauge(IdPtr id, Pub* publisher)
-      : StatelessMeter<Pub>(std::move(id), publisher) {}
+  Gauge(IdPtr id, Pub* publisher, unsigned int ttl = 0)
+      : StatelessMeter<Pub>(std::move(id), publisher), ttl(ttl) {
+    if (ttl > 0) {
+      type_str_ = "g," + std::to_string(ttl);
+    }
+  }
   void Set(double value) noexcept { this->send(value); }
 
  protected:
-  std::string_view Type() override { return "g"; }
+  std::string_view Type() override { return type_str_; }
+
+ private:
+  unsigned int ttl;
```

Comment:
Since we are storing the `type_str_` now, we don't need to keep a copy of the `ttl`.